modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc

```diff
@@ -0,0 +1,128 @@
+
+[[remove-the-encryption-records-from-the-owncloud-database]]
+== Remove the Encryption Records from the ownCloud Database
+
+Once your ownCloud files are unencrypted, and encryption has been disabled, you need to remove the encryption records from the database.
+There is, currently, no `occ` command to handle this, so it has to be done manually.
+Specifically, you need to remove all records from the `oc_appconfig` table where the `appid` column is set to `encryption`.
+
+In the examples below you can see how to do this using the SQLite database.
+If you are not using SQLite, please use the commands specific to your database vendor.
+
+[NOTE]
+====
+The example code assumes that the path to the SQLite database is `<YOUR/OWNCLOUD/ROOT/DIRECTORY>data/owncloud.database`.
+====
+
+[source,console]
+----
+sqlite3 data/owncloud.database
+sqlite> delete from oc_appconfig where appid='encryption';
+sqlite> select * from oc_appconfig where appid='encryption';
+----
+
+Same for mysql/mariadb:
+
+----
+SELECT * FROM `oc_appconfig` WHERE `appid` LIKE 'encryption'
+----
+
+[[remove-the-files_encryption-directory]]
+== Remove the `files_encryption` Directory
+
+With the database updated, next, the `files_encryption` directory needs to be removed.
+Below is an example of how to do so, to save you time.
+
+[source,console]
+----
+find ./data* -name files_encryption -exec rm -rvf {} \;
+----
+
+[[encrypt-the-filesystem-using-master-key-based-encryption]]
+== Encrypt the Filesystem Using Master Key-based Encryption
+
+Now, your ownCloud files can be encrypted using Master Key-based encryption.
+This requires the following steps:
+
+. The encryption app needs to be enabled
+. Encryption needs to be enabled
+. The encryption type needs to be set to master key
+. The ownCloud filesystem can be re-encrypted.
+
+The following example shows how to do this.
+
+[source,console]
+----
+sudo -u www-data php occ app:enable encryption && \
+  sudo -u www-data php occ encryption:enable && \
+  sudo -u www-data php occ encryption:select-encryption-type masterkey -y && \
+  sudo -u www-data php occ encryption:encrypt-all && \
+----
+
+[[verify-the-encrypted-files]]
+== Verify the Encrypted Files
+
+With the files encrypted using Master Key-based encryption, you should now verify that everything worked properly.
+To do so, run a `SELECT` query in your database which returns all files from the `oc_appconfig` table where the `appid` column is set to `encryption`.
+You should see a number of records, as in the output of the example below.
+
+[source,console]
+----
+$ sqlite3 data/owncloud.database
+sqlite> select * from oc_appconfig where appid='encryption';
+encryption|recoveryKeyId|recoveryKey_73facda6
+encryption|publicShareKeyId|pubShare_73facda6
+encryption|masterKeyId|master_73facda6
+encryption|installed_version|1.3.1
+encryption|types|filesystem
+encryption|enabled|yes
+encryption|useMasterKey|1
+----
+
+Same for mysql/mariadb:
```

Comment:
I know that it's rather pedantic, sorry, but please change "mysql/mariadb" to MySQL/MariaDB.

Reply:
no worries, I had a long break in working on the document and asciidoc in particular, still learning ,-)

Reply:
All good. Sorry if I'm being quite nitpicky. Not meaning to be painfful about this.